sympy/concrete/summations.py

```diff
@@ -1281,6 +1281,12 @@ def _eval_sum_hyper(f, i, a):
     if isinstance(hs, Float):
         from sympy.simplify.simplify import nsimplify
         hs = nsimplify(hs)
+    if hs.args:
+        for arg in hs.args:
+            if isinstance(arg, Float):
+                from sympy.simplify.simplify import nsimplify
+                hs = nsimplify(hs)
+                break
```

Comment:
I don't think that this change should be made. We shouldn't just convert a float into a rational. It's up to the user to do that if that's what they want.

Reply:
The conversion of a float to a rational was done initially so that further steps could be evaluated and a numerator/ denominator could be used. However, this particular case was not considered, where the number had one of the elements as a float and hence the issue arose. The final answer is still of a float type.

Reply:
Upon removing the float to rational conversion code, as follows: 
![image](https://user-images.githubusercontent.com/96943731/222040688-11dc9e04-cf89-4111-a42c-b5ceef924934.png)

This gives an error when testing the concrete module. The float to rational conversion was put in the first place to solve the issue #11642.

Reply:
Rather than converting to rational the better fix is just to identify where rational numbers and floats are being treated differently and then make sure that they are handled consistently. For #11642 the better fix is:
```diff
diff --git a/sympy/concrete/summations.py b/sympy/concrete/summations.py
index e4b1abe591..f688c7c07a 100644
--- a/sympy/concrete/summations.py
+++ b/sympy/concrete/summations.py
@@ -1278,16 +1278,12 @@ def _eval_sum_hyper(f, i, a):
     if hs is None:
         return None
 
-    if isinstance(hs, Float):
-        from sympy.simplify.simplify import nsimplify
-        hs = nsimplify(hs)
-
     from sympy.simplify.combsimp import combsimp
     from sympy.simplify.hyperexpand import hyperexpand
     from sympy.simplify.radsimp import fraction
     numer, denom = fraction(factor(hs))
-    top, topl = numer.as_coeff_mul(i)
-    bot, botl = denom.as_coeff_mul(i)
+    top, topl = numer.as_coeff_mul(i, rational=False)
+    bot, botl = denom.as_coeff_mul(i, rational=False)
     ab = [top, bot]
     factors = [topl, botl]
     params = [[], []]
```

Reply:
Thanks for the feedback, Oscar. I have made the suggested changes. I had to modify a test to allow an assertion test. The failing test: test / tests1 (3.12.0-alpha - 3.12, false) (pull_request) is something that I can't understand and I have seen it cause a problem in some other pull requests as well.